tests/test_skewing.py

```diff
@@ -154,25 +154,366 @@ def test_skewing_codegen(self, expr, expected, skewing, blockinner):
+        init_value = 0.2
+
+        # Field initialization
+        grid = Grid(shape=(nx, ny, nz))
+        u = TimeFunction(name='u', grid=grid, space_order=so)
+        u.data[:, :, :] = init_value
+
+        # Create an equation with second-order derivatives
+        a = Constant(name='a')
+        eq = Eq(u.dt, a*u.laplace, subdomain=grid.interior)
+        x, y, z = grid.dimensions
+        stencil = solve(eq, u.forward)
+        eq0 = Eq(u.forward, stencil)
+
+        op = Operator(eq0, opt=('advanced', {'openmp': True,
+                                'wavefront': False, 'blocklevels': 2}))
+
+        op.apply(time_M=nt, dt=dt, a=nu)
+        norm_u = norm(u)
+
+        u.data[:] = init_value
+
+        op1 = Operator(eq0, opt=('advanced', {'openmp': True,
+                                              'wavefront': True, 'blocklevels': 2}))
+
+        op1.apply(time_M=nt, dt=dt, a=nu)
+        assert np.isclose(norm(u), norm_u, atol=1e-4, rtol=0)
+
+        iters = FindNodes(Iteration).visit(op1)
+        time_iter = [i for i in iters if i.dim.is_Time]
+        assert len(time_iter) == 2
+        assert_structure(op, ['tx0_blk0y0_blk0x0_blk1y0_blk1xyz'])
+        assert_structure(op1, ['time0_blk0x0_blk0y0_blk0tx0_blk1y0_blk1xyz'])
+
+    @pytest.mark.parametrize('so, shape, nt',
+                             [(2, (15, 39, 21), 10), (2, (32, 16, 25), 13),
+                              (4, (18, 38, 38), 66), (4, (28, 18, 14), 45),
+                              (8, (17, 25, 42), 16), (8, (17, 25, 42), 17),
+                              (16, (13, 38, 34), 21), (16, (16, 35, 32), 19)])
+    def test_wave_correctness_IV(self, so, shape, nt):
+        nx, ny, nz = shape
+        nu = .5
+        dx = 2. / (nx - 1)
+        dy = 2. / (ny - 1)
+        dz = 2. / (nz - 1)
+        sigma = .25
+        dt = sigma * dx * dz * dy / nu
+
+        # Initialise u
+        init_value = 0.1
+
+        # Field initialization
+        grid = Grid(shape=(nx, ny, nz))
+        u = TimeFunction(name='u', grid=grid, space_order=so, time_order=2)
+        u.data[:, :, :] = init_value
+
+        # Create an equation with second-order derivatives
+        a = Constant(name='a')
+        eq = Eq(u.dt, a*u.laplace, subdomain=grid.interior)
+        x, y, z = grid.dimensions
+        stencil = solve(eq, u.forward)
+        eq0 = Eq(u.forward, stencil)
+
+        op = Operator(eq0, opt=('advanced', {'openmp': True,
+                                'wavefront': False, 'blocklevels': 2}))
+
+        op.apply(time_M=nt, dt=dt, a=nu)
+        norm_u = norm(u)
+
+        u.data[:] = init_value
+
+        op2 = Operator(eq0, opt=('advanced', {'openmp': True,
+                                              'wavefront': True, 'blocklevels': 2}))
+
+        op2.apply(time_M=nt, dt=dt, a=nu)
+        assert np.isclose(norm(u), norm_u, atol=1e-5, rtol=0)
+
+    @pytest.mark.parametrize('nt', [2, 24, 8, 17, 16, 19, 41])
+    @pytest.mark.parametrize('shape',
```

Comment:
"fuzzy testing" to cover corner cases for exotic min/max time/space block combos